src/Storages/ExternalStream/Kafka/KafkaSink.h

```diff
@@ -31,12 +31,13 @@ class ChunkPartitioner
 
     BlocksWithShard doParition(Block block, Int32 partition_cnt) const;
 
-    IColumn::Selector createSelector(const Block & block, Int32 partition_cnt) const;
+    IColumn::Selector createSelector(Block block, Int32 partition_cnt) const;
+
+    static inline std::minstd_rand rand{std::random_device()()};
```

Comment:
std::minstd_rand is usually not thread-safe, static object will be shared across different KafkaSink (hence likely multiple threads)

Reply:
As discussed, reverted it to per KafkaSink for now, and will introduce a global random generator later.